src/LiteCore/test/LiteCore.Tests.Shared/ScopeCollectionTest.cs

```diff
@@ -0,0 +1,667 @@
+        }
+
+        #endregion
+
+        #region Document
+
+        [Fact]
+        public void TestMoveDocWithinCollections()
+        {
+            var collName = "newColl";
+            var scopeName = "newScope";
+            
+            RunTestVariants(() =>
+            {
+                C4Document* doc = null;
+
+                using (var collName1_ = new C4String(collName + 1))
+                using (var scopeName1_ = new C4String(scopeName + 1))
+                using (var collName2_ = new C4String(collName + 2)) {
+                    var collectionSpecA = new C4CollectionSpec() {
+                        name = collName1_.AsFLSlice(),
+                        scope = scopeName1_.AsFLSlice()
+                    };
+
+                    var coll1 = (C4Collection*)LiteCoreBridge.Check(err =>
+                    {
+                        return Native.c4db_createCollection(Db, collectionSpecA, err);
+                    });
+
+                    var collectionSpecB = new C4CollectionSpec() {
+                        name = collName2_.AsFLSlice(),
+                        scope = scopeName1_.AsFLSlice()
+                    };
+
+                    var coll2 = (C4Collection*)LiteCoreBridge.Check(err =>
+                    {
+                        return Native.c4db_createCollection(Db, collectionSpecB, err);
+                    });
+
+                    LiteCoreBridge.Check(err => Native.c4db_beginTransaction(Db, err));
+                    try {
+                        doc = (C4Document*)LiteCoreBridge.Check(err => NativeRaw.c4coll_createDoc(coll1, DocID, FleeceBody, 0, err));
+                    } finally {
+                        LiteCoreBridge.Check(err => Native.c4db_endTransaction(Db, true, err));
+                        Native.c4doc_release(doc);
+                    }
+
+                    Native.c4coll_getDocumentCount(coll1).Should().Be(1, "Because coll1 should contain the doc that was just added.");
+                    Native.c4coll_getDocumentCount(coll2).Should().Be(0, "Because there is no doc in coll2.");
+
+                    //var moveDocSuccessful = (bool)LiteCoreBridge.Check(err =>
+                    //{
+                    //    /* Moves a document to another collection, possibly with a different docID.
+                    //    @param collection  The document's original collection.
+                    //    @param docID  The ID of the document to move.
+                    //    @param toCollection  The collection to move to.
+                    //    @param newDocID  The docID in the new collection, or a NULL slice to keep the original ID.
+                    //    @param error Information about any error that occurred
+                    //    @return True on success, false on failure. */
+                    //    return NativeRaw.c4coll_moveDoc(coll1, DocID, coll2, FLSlice.Null, err); //Exception thrown: read access violation. this was nullptr.
+                    //});
```

Comment:
Need @borrrden 's advice how to use c4coll_moveDoc properly, I am getting Exception thrown: read access violation. this was nullptr.

Reply:
I don't think there is a plan to use this API yet.  Also it's broken and there is a ticket for it already.

Reply:
This is a very handy method... Hope we will provide this in the future. Also removed for now..